resource reuse/plugins/org.polarsys.kitalpha.resourcereuse/src/org/polarsys/kitalpha/resourcereuse/registry/TargetPlatformReader.java

```diff
@@ -11,40 +11,140 @@
  *******************************************************************************/
 package org.polarsys.kitalpha.resourcereuse.registry;
 
-import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 
-import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IContributor;
+import org.eclipse.core.runtime.IExtension;
+import org.eclipse.core.runtime.IExtensionRegistry;
+import org.eclipse.core.runtime.RegistryFactory;
+import org.eclipse.core.runtime.spi.RegistryContributor;
 import org.eclipse.pde.core.plugin.IPluginModelBase;
-import org.eclipse.pde.core.plugin.PluginRegistry;
 import org.polarsys.kitalpha.resourcereuse.model.Location;
+import org.polarsys.kitalpha.resourcereuse.model.Resource;
 
 /**
+ * An implementation of the {@link ModelBasedReader} focusing on retrieving information focused on Platform bundles
+ * only.
+ * 
  * @author Thomas Guiu
+ * @author <a href="mailto:arnaud.dieumegard@obeo.fr">Arnaud Dieumegard</a>
  * 
  */
 public class TargetPlatformReader extends ModelBasedReader {
+    
+    private List<String> clearedBundles = new ArrayList<String>();
+
+    @Override
+    /*
+     * In this implementation we override readResources and thus do not need the content of the getModels method. The
+     * ModelBasedReader signature may need to be updated to allow to remove this method.
+     */
+    protected IPluginModelBase[] getModels() {
+        return new IPluginModelBase[0];
+    }
+
+    @Override
+    public ResourceRegistry readResources(ResourceRegistry parent) {
+        registry = new ModelBasedRegistry(parent, this);
+        // Ensure clearedBundles list is cleaned
+        clearedBundles.clear();
+        IExtensionRegistry extensionRegistry = RegistryFactory.getRegistry();
+
+        for (IConfigurationElement confElement : extensionRegistry.getConfigurationElementsFor(CONCERN_EXTENSION_ID)) {
```

Comment:
Why using two for loops (line 55 and line 68)?

Reply:
Because there are two ResourceReuse extensions (Resource and Concerns)